hazelcast/src/main/java/com/hazelcast/config/WanPublisherState.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+/**
+ * Defines the state in which a WAN publisher can be in if it is not shutting
+ * down.
+ */
+public enum WanPublisherState {
+    /**
+     * State where both enqueuing new events is allowed, enqueued events are
+     * replicated to the target cluster and WAN sync is enabled.
+     */
+    ACTIVE((byte) 0, true, true),
+    /**
+     * State where new events are enqueued but they are dequeued. Some events
+     * which have been dequeued before the state was switched may still be
+     * replicated to the target cluster but further events will not be
+     * replicated. WAN sync is enabled.
+     */
+    PAUSED((byte) 1, true, false),
+    /**
+     * State where neither new events are enqueued nor dequeued. As with the
+     * {@link #PAUSED} state, some events might still be replicated after the
+     * publisher has switched to this state. WAN sync is enabled.
+     */
+    STOPPED((byte) 2, false, false);
+
+    private final boolean enqueueNewEvents;
+    private final boolean replicateEnqueuedEvents;
+    private final byte id;
+
+    WanPublisherState(byte id,
+                      boolean enqueueNewEvents,
+                      boolean replicateEnqueuedEvents) {
+        this.id = id;
+        this.enqueueNewEvents = enqueueNewEvents;
+        this.replicateEnqueuedEvents = replicateEnqueuedEvents;
+    }
+
+    /**
+     * Returns the WanPublisherState as an enum.
+     */
+    public static WanPublisherState getByType(final byte id) {
+        for (WanPublisherState state : values()) {
```

Comment:
we can cache result of `values()` to not to create an array on each access.

Reply:
Done.